### ISSUE TITLE
Augment EOL Check to catch missing newline at EOF

### ISSUE THREAD

**Issue by dilijev** [Microsoft/ChakraCore#496]

**Comment by s4sarishma** [comment]
Hi @dilijev, Are you still looking for a fix for this issue?

**Comment by dilijev** [comment]
@srs88 Sure! Want to take a crack at it?

**Comment by srs88** [comment]
Yes! I am a student trying to learn the codebase. Can you give me some pointers to start with?

**Comment by dilijev** [comment]
For this one you'll basically just need to look at 
https://github.com/Microsoft/ChakraCore/tree/master/jenkins 

Files: check_*.sh

You just need to follow the patterns there, come up with a way to detect whether each (non-excluded) file has the last character in the file as a line feed (0x0a). This issue suggests such a check would make sense as additional logic in `check_file_eol.sh`

The list of files (and excluded files) are computed here: https://github.com/Microsoft/ChakraCore/blob/master/jenkins/check_eol.sh#L18
and the list is sent to `check_file_eol.sh` (one at a time).

**Comment by dilijev** [comment]
When you submit a PR, your check will run automatically because Jenkins is already configured to run `check_eol.sh` as a task.

As part of your change you'll have to correct any files which are incorrect so that you pass your own new check! You'll also have confirmation about whether it's working.

Please try to verify your check by running it on your own computer first.

**Comment by srs88** [comment]
Thank you for the tips. I will try and get back to you.

**Comment by dilijev** [comment]
@srs88 wondering if you've had any success?

**Comment by srs88** [comment]
Hi @dilijev. My apologies. I got busy with my course work and missed this. I will try this today.

**Comment by dilijev** [comment]
@srs88 no pressure! :) Let us know if you need any help. Feel free to reach out on https://gitter.im/Microsoft/ChakraCore

**Comment by rhuanjl** [comment]
I've made a PR to respond to this #5588

**Comment by MSLaguana** [comment]
I think you mean https://github.com/Microsoft/ChakraCore/pull/5583

**Comment by rhuanjl** [comment]
@MSLaguana indeed, sorry for the typo and thanks for the correction (fixed above)
